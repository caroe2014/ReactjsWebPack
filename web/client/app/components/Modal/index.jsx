// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React, { Component} from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Modal as ModalWindow, Flex, Fixed, Heading, Text } from 'rebass';
import PrimaryButton from 'web/client/app/components/PrimaryButton';
import SecondaryButton from 'web/client/app/components/SecondaryButton';
import IconButton from 'web/client/app/components/IconButton';
import i18n from 'web/client/i18n';

const ModalContainer = styled(ModalWindow)`
  border-radius: 4px;
  z-index: 1200;
  width: 420px;
  padding: 15px;
  ${props => props.theme.mediaBreakpoints.mobile`
    ${props => { return props.menuChangePopup ? `width: 100%; height: 100%;` : 'width: 300px;'; }};
`};
`;

const ModalBackground = styled(Fixed)`
  background-color: rgba(0, 0, 0, 0.5);
  z-index: 1200;
  top: 0px;
  right: 0px;
  bottom: 0px;
  left: 0px;
  opacity: 1;
  will-change: opacity;
  transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
`;

const ModalTitle = styled(Heading)`
  margin: 38px 0 30px;
  color: ${props => props.theme.colors.primaryTextColor};
  font-size: 1.5em;
  font-weight: 300;
  text-align: center;
  &:focus {
    outline: none;
    box-shadow: none;
  }
`;

const ModalText = styled(Text)`
  color: ${props => props.theme.colors.secondaryTextColor};
  font-size: 1em;
  text-align: center;
  padding: 0px 10px;
  &:focus {
    outline: none;
    box-shadow: none;
  }
`;

const ModalFooter = styled(Flex)`
  flex-direction: row;
  justify-content: center;
  margin-top: 35px;
  ${props => {
    return props.menuChangePopup && `
    flex-direction: column;
    align-items: center;
    width: 100%;
    .modal-cancel-button, .modal-continue-button {
      width: 80%;
      background-color: ${props.theme.colors.buttonControlColor};
      color: ${props.theme.colors.buttonTextColor};
      margin-bottom: 10px;
    }
  `;
  }};
`;

const MainButton = styled(props => <PrimaryButton {...props} />)`
  border-radius: 4px;
  font-size: 0.8rem;
  margin: 5px;
  padding: 15px 0;
  text-align: center;
  width: 140px;
  &:focus {
    outline: none;
    box-shadow: none;
  }
`;

const CancelButton = styled(props => <SecondaryButton {...props} />)`
  border-radius: 4px;
  font-size: 0.8rem;
  margin: 5px;
  padding: 15px 0;
  text-align: center;
  width: 140px;
  &:focus {
    outline: none;
    box-shadow: none;
  }
`;

const CloseButton = styled(props => <IconButton {...props} />)`
  color: ${props => props.theme.colors.textGrey};
  font-size: ${props => props.theme.fontSize.md};
  font-weight: 300;
  position: absolute;
  top: 16px;
  &:focus {
    outline: none;
    box-shadow: none;
  }
`;

class Modal extends Component {

  constructor (props) {
    super(props);
    this.makeDialogAloneAccessible = this.makeDialogAloneAccessible.bind(this);
    this.onContinue = this.onContinue.bind(this);
  }

  componentDidUpdate (prevProps) {
    if (!prevProps.open && this.props.open) {
      this.makeDialogAloneAccessible();
      document.getElementById('modal-title') && document.getElementById('modal-title').focus();
    }
  }

  makeDialogAloneAccessible () {
    // Accessibility for dialog
    const parent = document.querySelector('.parent');
    const children = parent.children;
    Array.from(children).forEach(child => {
      if (child.id !== 'parent-modal') {
        child.inert = true;
      }
    });
  }

  onContinue () {
    this.props.onContinue(this.props.modalName);
  }

  render () {
    const { open, onEscape, title, textI18nKey, dynamicKey, text, showCancelButton, onCancel, cancelButtonText,
      continueButtonText, showClose, onClose, menuChangePopup } = this.props;
    return (
      <React.Fragment>
        {open && (
          <Flex id='parent-modal' role='alert'>
            <ModalBackground />
            <ModalContainer className='modal-container' onKeyDown={onEscape} menuChangePopup={menuChangePopup}>
              <ModalTitle className='modal-title' tabIndex={0} id='modal-title'
                aria-label={title}>{title}</ModalTitle>
              <ModalText className='modal-content'>
                {textI18nKey ? i18n.t(textI18nKey, dynamicKey).split('\n').map((item, i) => (
                  <Text className='modal-text' key={i} tabIndex={0} aria-label={item}>{item}</Text>
                ))
                  : text.split('\n').map((item, i) => (
                    <Text key={i} tabIndex={0} aria-label={item}>{item}</Text>
                  ))}
              </ModalText>
              <ModalFooter className='modal-footer' menuChangePopup={menuChangePopup}>
                {showCancelButton && (
                  <CancelButton
                    className='modal-cancel-button'
                    onClick={onCancel}
                    tabIndex={0}
                    aria-label={cancelButtonText}>
                    {cancelButtonText}
                  </CancelButton>
                )}
                <MainButton
                  className='modal-continue-button'
                  onClick={this.onContinue}
                  tabIndex={0}
                  ariaLabel={continueButtonText}>
                  {continueButtonText}
                </MainButton>
              </ModalFooter>
              {showClose &&
              <CloseButton className='close-button' tabIndex={0}
                aria-label={i18n.t('EXIT_ALERT')} children='&#10005;' onClick={onClose || onCancel} />
              }
            </ModalContainer>
          </Flex>
        )}
      </React.Fragment>
    );
  }
}

Modal.propTypes = {
  open: PropTypes.bool,
  continueButtonText: PropTypes.string.isRequired,
  cancelButtonText: PropTypes.string,
  showCancelButton: PropTypes.bool,
  onContinue: PropTypes.func.isRequired,
  onCancel: PropTypes.func.isRequired,
  title: PropTypes.string,
  text: PropTypes.string
};

export default Modal;
