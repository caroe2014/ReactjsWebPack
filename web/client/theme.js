// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import { css, injectGlobal } from 'styled-components';

const theme = {
  colors: {
    black: '#000000',
    navigationMenu: '#48a942',
    navigationMenuHover: 'cornflowerblue',
    primary: '#347CFF',
    primaryButton: '#347CFF',
    loginField: '#AAA7AC',
    placeHolder: '#909090',
    loginBoxColor: '#434149',
    open: '#48a942',
    close: 'red',
    footer: '#48a942',
    textGrey: '#505050',
    paymentSuccessTitle: '#505050',
    inputBottomBorder: 'lightgray',
    backgroundGrey: '#E6E6E6',
    text: 'black',
    primaryText: '#383838',
    secondaryText: '#a6a6a6',
    hint: '#a6a6a6',
    light: 'white',
    error: '#cc0000',
    tile: '#505050',
    addToCart: '#347CFF',
    itemTile: '#3C474C0',
    darkGrey: '#464646',
    itemBorder: 'lightgrey',
    addOnTab: '#347CFF',
    validationError: '#cc0000',
    loginBackground: '#303030',
    loginFormBackground: '#484848',
    disableBackground: '#E4E4E4',
    addonBg: '#EEEEEE',
    glutenFree: '#8d7356',
    spicy: '#ed1c24',
    vegan: '#0d4a22',
    vegetarian: '#0d4a22',
    new: '#ff6000',
    healthy: '#359919',
    contextBackground: '#f0f0f3',
    loader: '#ababab',
    disableButton: '#D1D1D1'
  },
  fontFamily: 'Roboto',
  receiptFontFamily: 'Roboto',
  fontSize: {
    xl: '2.4rem',
    lg: '1.8rem',
    md: '1.3rem',
    nm: '1rem',
    sm: '0.75rem'
  },
  fontSizeFromProps: (props) => {
    if (props.xl) return props.theme.fontSize.xl;
    if (props.lg) return props.theme.fontSize.lg;
    if (props.md) return props.theme.fontSize.md;
    if (props.nm) return props.theme.fontSize.nm;
    if (props.sm) return props.theme.fontSize.sm;
    return 'inherit';
  },
  spacing: {
    xl: '8em',
    lg: '4em',
    md: '2em',
    nm: '1em',
    sm: '0.5em',
    none: '0em'
  },
  mediaBreakpoints: {
    mobile: (...args) => css`@media (max-width: 32em) {${css(...args)}}`,
    tablet: (...args) => css`@media (max-width: 72em) {${css(...args)}}`,
    desktop: (...args) => css`@media (min-width: 72em) {${css(...args)}}`
  },
  shadow: '#cfcfcf 0.1px 0.1px 3px 0px',
  nav: {
    height: '70px'
  }
};

injectGlobal`
  .agilysys-icon-new {
    color: ${theme.colors.new}
  }
  .agilysys-icon-healthy {
    color: ${theme.colors.healthy}
  }
  .agilysys-icon-vegetarian {
    color: ${theme.colors.vegetarian}
  }
  .agilysys-icon-vegan {
    color: ${theme.colors.vegan}
  }
  .agilysys-icon-spicy {
    color: ${theme.colors.spicy}
  }
  .agilysys-icon-gluten-free {
    color: ${theme.colors.glutenFree}
  }
  @media print {
    @-moz-document url-prefix() {
      .appContainer, .appContainer>div, .appContainer .parent, .appContainer .BottomContainer {
        display: block !important;
      }
    }
  }
`;

export default theme;
