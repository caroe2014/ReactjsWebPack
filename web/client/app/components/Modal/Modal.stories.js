// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React from 'react';

import { storiesOf } from '@storybook/react';
import { withKnobs } from '@storybook/addon-knobs/react';
import { Provider } from 'rebass';
import theme from 'web/client/theme.js';
import Modal from '.';

storiesOf('Modal', module)
  .addDecorator(withKnobs)
  .add('Error Modal', () => {
    return (
      <div>
        <Provider theme={theme}>
          <Modal
            open
            cancelButtonText={'Cancel'}
            continueButtonText={'Got it'}
            title={'Oops!'}
            text={'Something went wrong \nwe are working on it, please try again after few minutes'}
          />
        </Provider>
      </div>
    );
  })
  .add('Confirmation Modal', () => {
    return (
      <div>
        <Provider theme={theme}>
          <Modal
            open
            showCancelButton
            cancelButtonText={'Cancel'}
            continueButtonText={'Continue'}
            title={'Title!'}
            text={'Some Text'}
          />
        </Provider>
      </div>
    );
  });
