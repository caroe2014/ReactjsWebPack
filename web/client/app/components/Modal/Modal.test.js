// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React from 'react';
import {
  shallow,
  mount,
  configure
} from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import {
  ThemeProvider
} from 'styled-components';
import theme from 'web/client/theme';
import { t } from 'i18next';

import Modal from './index';
configure({
  adapter: new Adapter()
});

jest.mock('i18next');
t.mockImplementation((key) => {
  return key;
});

jest.mock('react-i18next', () => ({
  Trans: 'div'
}));

jest.mock('web/client/i18n', () => ({
  t: (key) => {
    return key;
  }
}));

describe('Modal', () => {

  it('does a snapshot test', () => {
    const wrapper = shallow(<Modal continueButtonText='' showClose text={'test'} showCancelButton onContinue={jest.fn()} onCancel={jest.fn()} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render without throwing an error', () => {
    expect(shallow(<Modal continueButtonText='' text={'test'} showClose showCancelButton onContinue={jest.fn()} onCancel={jest.fn()} />).exists()).toBe(true);
  });

  it('should render without throwing an error on mount', () => {
    const mountWrapper = mount(
      <ThemeProvider theme={theme}>
        <Modal open continueButtonText='' text={'test'} showClose textI18nKey={'test'} showCancelButton onContinue={jest.fn()} onCancel={jest.fn()} />
      </ThemeProvider>
    );
    expect(mountWrapper.exists()).toBe(true);
  });

  it('should render without throwing an error on mount', () => {
    const mountWrapper = mount(
      <ThemeProvider theme={theme}>
        <Modal open continueButtonText='' text={'test'} showClose showCancelButton onContinue={jest.fn()} onCancel={jest.fn()} />
      </ThemeProvider>
    );
    expect(mountWrapper.exists()).toBe(true);
  });
});
