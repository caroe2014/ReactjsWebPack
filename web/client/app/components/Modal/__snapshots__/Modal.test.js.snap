// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal does a snapshot test 1`] = `<Fragment />`;
